lib/messages.js

```diff
@@ -827,6 +829,127 @@ util.inherits(GetHeaders, GetBlocks);
 module.exports.GetHeaders = Message.COMMANDS.getheaders = GetHeaders;
 
 /**
+ * Request peer to apply a bloom filter to 'inv' messages sent back
+ *
+ * @name P2P.Message.filterload
+ * @param{BloomFilter} filter - a BloomFilter object
+ */
+function FilterLoad(filter) {
+  this.command = 'filterload';
+  $.checkArgument(_.isUndefined(filter) || filter instanceof BloomFilter,
+    'Object or undefined required for filter object');
+
+  // TODO: Are these defaults (1,0.01) sane?
+  this.filter = filter || BloomFilter.create(1, 0.01);
+  return this;
+}
+util.inherits(FilterLoad, Message);
+
+FilterLoad.prototype.fromBuffer = function(payload) {
+  $.checkArgument(payload);
+  var parser = new BufferReader(payload);
+  var data = parser.readVarLengthBuffer();
+  $.checkState(data.length <= BloomFilter.MAX_BLOOM_FILTER_SIZE,
+    'Filter data must be <= MAX_BLOOM_FILTER_SIZE bytes');
+  var nHashFuncs = parser.readUInt32LE();
+  $.checkState(nHashFuncs <= BloomFilter.MAX_HASH_FUNCS,
+    'Filter nHashFuncs must be <= MAX_HASH_FUNCS');
+  var nTweak = parser.readUInt32LE();
+  var nFlags = parser.readUInt8();
+  this._checkFinished(parser);
+
+  var vData = [];
+  var dataParser = new BufferReader(data);
+  for(var i; i < data.length; i++) {
+    vData[i] = dataParser.readUInt8();
+  }
+  this.filter = new BloomFilter({
+    vData: vData,
+    nHashFuncs: nHashFuncs,
+    nTweak: nTweak,
+    nFlags: nFlags
+  });
+
+  return this;
+};
+
+FilterLoad.prototype.toBloomFilter = function() {
```

Comment:
Don't think this is necessary, since `this.filter` is an instance of BloomFilter